src/components/profiles/ProfileStatusToggle.js

```diff
@@ -0,0 +1,94 @@
+import React, { useContext, useState } from "react";
+import { Button, Col, Modal, Row } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+import { ProfileContext } from "./ProfileProvider";
+
+export default (props) => {
+  const { userId, isStaff } = props;
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [show, setShow] = useState(false);
+  const history = useHistory();
+  const currentRareUserId = parseInt(localStorage.getItem("rare_user_id"));
+
+  const { updateUserRole } = useContext(ProfileContext);
+
+  const handleStatusToggle = (e) => {
+    // check to see if current user (an admin) is choosing to demote themselves, and if so render a modal
+    // prompting them to confirm; otherwise, toggle selected user's status
+    if (currentRareUserId === userId && e.target.value === "false") {
+      setShow(true);
+    } else {
+      setIsSubmitting(true);
+      const isStaffValue = e.target.value;
+      updateUserRole(userId, { is_staff: isStaffValue }).then(() =>
+        setIsSubmitting(false)
+      );
+    }
+  };
+
+  //runs when a user (admin) confirms self-demotion; re-pushing the path to the profiles list is necessary to re-render
+  // after removing the "is_admin" token
+  const handleSelfNuke = (e) => {
```

Comment:
also what an excellent touch, this whole prompt the user they're gonna nuke themselves and require confirmation AND destroy the localStorage token and redirect them so their view updates all so good